### ISSUE TITLE
Exception: Failed to parse URL from

### ISSUE THREAD

**Issue by miroslavnavratil** [pnp/pnpjs#2624]
### Category
- [ ] Question

### Version
2.15.0

Please specify what version(s) of SharePoint you are targeting: [     Sharepoint Online    ]

### Expected / Desired Behavior / Question
Tried to make a sample call to Sharepoint Online from Azure Function as described here https://pnp.github.io/pnpjs/v2/nodejs-support/ 

import { AzureFunction, Context, HttpRequest } from "@azure/functions"
import { SPFetchClient } from "@pnp/nodejs-commonjs";
import { sp } from "@pnp/sp-commonjs";

const httpTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
    context.log('HTTP trigger function processed a request.');

    sp.setup({
        sp: {
            fetchClientFactory: () => {
                return new SPFetchClient("https://xxxxxx.sharepoint.com/", "xxxxxxx", "xxxxxxxx");
            },
        },
    });

    const w = await sp.web();
    console.log(JSON.stringify(w, null, 2));

    const name = (req.query.name || (req.body && req.body.name));
    const responseMessage = name
        ? "Hello, " + name + ". This HTTP triggered function executed successfully."
        : "This HTTP triggered function executed successfully. Pass a name in the query string or in the request body for a personalized response.";

    context.res = {
        // status: 200, /* Defaults to 200 */
        body: responseMessage
    };

};

export default httpTrigger;

### Observed Behavior
Always end up with the following error

Exception: Failed to parse URL from
Stack: TypeError: Failed to parse URL from

Here is the callstack

[2023-03-31T13:55:02.005Z] Executed 'Functions.HttpTrigger1' (Failed, Id=749c5531-8274-42bc-9e7f-6710f962ac95, Duration=120ms)
[2023-03-31T13:55:02.008Z] System.Private.CoreLib: Exception while executing function: Functions.HttpTrigger1. System.Private.CoreLib: Result: Failure
Exception: Failed to parse URL from
Stack: TypeError: Failed to parse URL from
    at new Request (node:internal/deps/undici/undici:7004:19)
    at mergeHeaders (D:\Mirek\VSCode\AFT\node_modules\@pnp\common-commonjs\net.js:9:20)
    at SPHttpClient.<anonymous> (D:\Mirek\VSCode\AFT\node_modules\@pnp\sp-commonjs\sphttpclient.js:38:51)
    at step (D:\Mirek\VSCode\AFT\node_modules\tslib\tslib.js:143:27)
    at Object.next (D:\Mirek\VSCode\AFT\node_modules\tslib\tslib.js:124:57)
    at D:\Mirek\VSCode\AFT\node_modules\tslib\tslib.js:117:75
    at new Promise (<anonymous>)
    at __awaiter (D:\Mirek\VSCode\AFT\node_modules\tslib\tslib.js:113:16)
    at SPHttpClient.fetch (D:\Mirek\VSCode\AFT\node_modules\@pnp\sp-commonjs\sphttpclient.js:28:38)
    at D:\Mirek\VSCode\AFT\node_modules\@pnp\odata-commonjs\pipeline.js:167:24.

**Comment by bcameron1231** [comment]
Possible Duplicate. #2461

Can you confirm what version of node you are running? Additionally, any particular reason why you're still using the V2 version of PnPjs and not using the latest V3 version?

**Comment by miroslavnavratil** [comment]
Let me thank you first for you fast feedback. You are right, it's a duplicate. Sorry for that.

I have downgraded my node.js to v16.20.0 and it's working now.

Regarding your question: my understanding is that V3 is more or less under development and not published yet thus I decided to use V2.

And let me have a question: could you kindly advice me where to find supported node.js version by a given library?

**Comment by bcameron1231** [comment]
V3 is not under development, it's fully published and available for you to use! Check out our [documentation ](https://pnp.github.io/pnpjs/getting-started/) on how to start using it! (be aware, it's not under the @pnp/pnpjs npm module)

As for the node version, I'm not sure if we have documented the V2 node version. Node 18 wasn't out when V2 was released, so it wasn't a problem at the time. We do document the V3 version to be >= 14, and that does work on Node 18.
